UpsertSpecialServiceActivationActivity/handler.ts

```diff
@@ -51,28 +75,45 @@ const upsertServiceActivation = (
 export const getUpsertSpecialServiceActivationActivityHandler = (
   servicesClient: ServicesAPIClient,
   logPrefix: string = "UpsertSpecialServiceActivationActivity"
-) => (context: Context, input: unknown): Promise<ActivityResult> => {
-  const fail = failure(context, logPrefix);
-  return pipe(
+) => (
+  context: Context,
+  input: unknown
+): Promise<Failure | ActivityResultSuccess> =>
+  pipe(
     input,
     ActivityInput.decode,
     TE.fromEither,
-    TE.mapLeft(errs =>
-      fail(errorsToError(errs), "Cannot decode Activity Input")
-    ),
+    TE.mapLeft(flow(errorsToError, permanentFailure)),
     TE.chain(activityInput =>
-      pipe(
-        upsertServiceActivation(
-          servicesClient,
-          activityInput.activationStatus,
-          activityInput.fiscalCode
-        ),
-        TE.bimap(
-          err => fail(err, "Cannot upsert special service activation"),
-          () => success()
-        )
+      upsertServiceActivation(
+        servicesClient,
+        activityInput.activationStatus,
+        activityInput.fiscalCode
       )
     ),
+    TE.bimap(
+      err => {
+        const error = TransientFailure.is(err)
+          ? `${logPrefix}|TRANSIENT_ERROR=${err.reason}`
+          : `${logPrefix}|FATAL|PERMANENT_ERROR=${err.reason}`;
+        trackException({
+          exception: new Error(error),
+          properties: {
+            // In case the the input (message from queue) cannot be decoded
+            // we mark this as a FATAL error since the lock on user's family won't be relased
+            detail: err.kind,
+            fatal: PermanentFailure.is(err).toString(),
+            isSuccess: false,
+            name: "cgn.exception.upsertSpecialService.failure"
+          }
+        });
+        context.log.error(error);
+        if (PermanentFailure.is(err)) {
+          throw new Error(err.reason);
+        }
+        return err;
```

Comment:
Permanent failure should return and transient should throw.
```suggestion
        if (TransientFailure.is(err)) {
          throw new Error(err.reason);
        }
        return err;
```